examples/cpp/pyperf/PyPerfNativeStackTrace.cc

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) Granulate. All rights reserved.
+ * Licensed under the AGPL3 License. See LICENSE.txt for license information.
+ */
+
+#include "PyPerfNativeStackTrace.h"
+
+#include <sys/uio.h>
+#include <unistd.h>
+
+#include <cstdio>
+#include <cstring>
+
+#include "PyPerfLoggingHelper.h"
+
+namespace ebpf {
+namespace pyperf {
+
+const uint8_t *NativeStackTrace::stack = NULL;
+size_t NativeStackTrace::stack_len = 0;
+uint64_t NativeStackTrace::sp = 0;
+uint64_t NativeStackTrace::ip = 0;
+
+NativeStackTrace::NativeStackTrace(uint32_t pid, const unsigned char *raw_stack,
+                                   size_t stack_len, uint64_t ip, uint64_t sp) : errors(0) {
+  NativeStackTrace::stack = raw_stack;
+  NativeStackTrace::stack_len = stack_len;
+  NativeStackTrace::ip = ip;
+  NativeStackTrace::sp = sp;
+
+  unw_accessors_t my_accessors = _UPT_accessors;
+  my_accessors.access_mem = NativeStackTrace::access_mem;
+  my_accessors.access_reg = NativeStackTrace::access_reg;
+
+  unw_addr_space_t as = unw_create_addr_space(&my_accessors, 0);
+  void *upt = _UPT_create(pid);
```

Comment:
Creating it for every sample means it will re-read `/proc/pid/maps` and re-open the DSOs, etc.

Please give an estimate on its effect (I'd say: run PyPerf on a system with some Python processes taking CPU; record PyPerf with perf and compare `master` with your branch). If effect grows too large, we might want to add caching here.

Reply:
The overhead isn't here. The heavy functions are `unw_get_proc_name` and `unw_step` in the constructor (the unwinding itself).